### PR TITLE
Add IBM Power VS: tfvars

### PR DESCRIPTION
**Pull request by clnperez**
For more background on IPI on Power VS, refer to the enhancement
proposal here: openshift/enhancements#736

Older discussions on some of the code here can be found in #5224

Signed-off-by: Christy Norman <christy@linux.vnet.ibm.com>

### CODE REVIEW

File: pkg/tfvars/powervs/powervs.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
+// Package powervs contains Power Virtual Servers-specific Terraform-variable logic.
+package powervs
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/rand"
+	"time"
+
+	"github.com/openshift/cluster-api-provider-powervs/pkg/apis/powervsprovider/v1alpha1"
```
Comment (Prashanth684) [inline pkg/tfvars/powervs/powervs.go:10]: machine-api-provider-powervs now ?

**Comment by clnperez** [comment]
working on cleaning this up now that the others are in. got sidetracked

File: pkg/asset/cluster/tfvars.go [hunk 2]
```diff
@@ -672,6 +675,46 @@ func (t *TerraformVariables) Generate(parents asset.Parents) error {
...
+
+		data, err = powervstfvars.TFVars(
+			powervstfvars.TFVarsSources{
+				MasterConfigs:        masterConfigs,
+				Region:               installConfig.Config.Platform.PowerVS.Region,
+				Zone:                 installConfig.Config.Platform.PowerVS.Zone,
+				APIKey:               os.Getenv("IC_API_KEY"),
+				SSHKey:               installConfig.Config.SSHKey,
+				PowerVSResourceGroup: installConfig.Config.PowerVS.PowerVSResourceGroup,
+				ImageBucketFileName:  string(*rhcosImage),
```
[the remark this answers is not in the document]
Comment (clnperez) [reply pkg/asset/cluster/tfvars.go:703]: oops. i should've included that file. adding here. thanks @Prashanth684 !

**Comment by clnperez** [comment]
tests seem stuck since yesterday
/retest

**Comment by clnperez** [comment]
i fixed the tf formatting. can you re-lgtm this @rna-afk ?

**Comment by rna-afk** [comment]
Could you move the rhcos changes to the first commit? I think just two commits are fine with one for vendoring and other for the code.

**Comment by clnperez** [comment]
done @rna-afk

**Comment by clnperez** [comment]
blergh. i'm out until tuesday (starting tomorrow) @patrickdillon could you approve this one?

and @Prashanth684 would you mind babysitting (/retest-required) if those keep failing?

File: pkg/asset/cluster/tfvars.go [hunk 3]
```diff
@@ -672,6 +675,46 @@ func (t *TerraformVariables) Generate(parents asset.Parents) error {
...
+		for i, m := range masters {
+			masterConfigs[i] = m.Spec.ProviderSpec.Value.Object.(*powervsprovider.PowerVSMachineProviderConfig)
+		}
+
+		data, err = powervstfvars.TFVars(
+			powervstfvars.TFVarsSources{
+				MasterConfigs:        masterConfigs,
+				Region:               installConfig.Config.Platform.PowerVS.Region,
+				Zone:                 installConfig.Config.Platform.PowerVS.Zone,
+				APIKey:               os.Getenv("IC_API_KEY"),
```
Comment (patrickdillon) [inline pkg/asset/cluster/tfvars.go:700]: You shouldn't set an environment variable to retrieve that value later in the code... This API key is in your session struct. The session struct should be stored in the install config metadata and then you can use that to retrieve the value here.

**Comment by clnperez** [comment]
just picked up the new nutanix provider. should be good again.

**Comment by clnperez** [comment]
aaaaand there's a new conflict. this will be fun :D

**Comment by clnperez** [comment]
@rna-afk i think that's resolved now as well so will need that lgtm again

**Pull request merged**